internal/workflow.go

```diff
@@ -66,8 +66,20 @@ var (
 )
 
 type (
+	// commonChannel contains methods common to both SendChannel and ReceiveChannel
+	commonChannel interface {
+		// Name returns the name of the Channel.
+		// If the Channel was retrieved from a GetSignalChannel call, Name returns the signal name.
+		//
+		// A Channel created without an explicit name will use a generated name by the SDK and
+		// is not deterministic.
+		Name() string
+	}
+
 	// SendChannel is a write only view of the Channel
 	SendChannel interface {
+		commonChannel
```

Comment:
I wonder if, even though we use aliases which make this harder, inlining the `Name()` method with the docs is clearer than a shared interface. It's just two places.

Reply:
Will make the change if there's agreement between @Quinn-With-Two-Ns and yourself :)

One readability benefit of sharing the interface is it'll ensure documentation remains synced over time if it gets modified

Reply:
Yes I think inlining is better for readability

Reply:
Updated and inlined. Also made tests explicitly test `ReceiveChannel` and `SendChannel` interfaces with Name